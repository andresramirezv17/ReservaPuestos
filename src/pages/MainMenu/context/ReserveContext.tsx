import React, { createContext, useEffect, useState } from 'react';
import moment from 'moment';
import { SearchParameters } from './models/SearchParameters';
import { WorkspacesSearched } from './models/WorkspacesSearched';
import { Floors } from './models/Floors';
import { Sections } from './models/Sections';
import * as ReserveServices from '../services/WorkspaceServices';

export interface WorkSpaceParameter {
  workspacesSearched?: WorkspacesSearched;
  floors?: Floors;
  sections?: Sections;
}

export interface propsSection {
  idFloor: number;
  idBuilding: number;
  name: string;
}

export const useStateContainer = (initialState: WorkSpaceParameter) => {
  const [workplaces, setworkplaces] = useState(
    initialState.workspacesSearched || { workplaces: [] },
  );

  const [floors, setfloors] = useState(
    initialState.floors || { floorsResult: [] },
  );
  const [sections, setsections] = useState(
    initialState.sections || { sectionsResult: [] },
  );

  const [parameters, setparameters] = useState('');
  const [building, setbuilding] = useState('Torre Central 2');
  const [floor, setfloor] = useState('Piso 1');
  const [section, setsection] = useState('Seccion A');
  const [date, setdate] = useState(
    moment(new Date()).add(1, 'days').format('DD/MM/YYYY'),
  );

  const searchWorkplaces = () => {
    ReserveServices.getSearchedWorkplaces().then((resultSearch) => {
      setworkplaces(resultSearch);
    });
  };
  const obtainFloors = () => {
    ReserveServices.getFloors().then((resultFloor) => setfloors(resultFloor));
  };

  const obtainSections = (id?: number) => {
    ReserveServices.getSections().then((resultSections) => {
      setsections(resultSections);
    });
  };

  const changeParameter = (parameter: string, value: string) => {
    if (parameter === 'piso') {
      setfloor(value);
    }
    if (parameter === 'section') {
      setsection(value);
    }
    if (parameter === 'date') {
      setdate(value);
    }
  };

  return {
    data: { workplaces, floors, sections, building, floor, section, date },
    mutations: {
      searchWorkplaces,
      obtainFloors,
      obtainSections,
      changeParameter,
    },
  };
};

export const ReserveContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface SearchWorkspaceProvider {
  initialState?: WorkSpaceParameter;
}

export const ReserveProvider: React.FC<SearchWorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <ReserveContext.Provider value={contextValue}>
      {children}
    </ReserveContext.Provider>
  );
};
