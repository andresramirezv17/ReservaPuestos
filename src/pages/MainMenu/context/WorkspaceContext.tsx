import React, { createContext, useEffect, useState } from 'react';
import { UserWorkspace } from './models/UserWorkspace';
import * as WorkspaceService from '../services/WorkspaceServices';

export interface WorkSpaceState {
  workSpace?: UserWorkspace;
  varpueba?: boolean;
}

export const useStateContainer = (initialState: WorkSpaceState) => {
  const [workspaceUser, setworkspaceUser] = useState(
    initialState.workSpace || {},
  );
  const [prueba, setprueba] = useState(initialState.varpueba || true);

  useEffect(() => {
    WorkspaceService.getReserves(1).then((reserves) =>
      setworkspaceUser(reserves),
    );
  }, []);

  return { workspaceUser, prueba };
};

export const WorkspaceContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface WorkspaceProvider {
  initialState?: WorkSpaceState;
}

export const UserWorkspaceProvider: React.FC<WorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <WorkspaceContext.Provider value={contextValue}>
      {children}
    </WorkspaceContext.Provider>
  );
};
