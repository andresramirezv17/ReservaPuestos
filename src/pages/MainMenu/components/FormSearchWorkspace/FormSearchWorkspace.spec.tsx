import React from 'react';
import { FormSearchWorkspace } from '.';
import { render, fireEvent } from '@testing-library/react';
import { ReserveProvider } from '../../context/ReserveContext';

describe('FormSearch test', () => {
  it('should match with snapshot', () => {
    const { container } = render(
      <ReserveProvider>
        <FormSearchWorkspace />
      </ReserveProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
