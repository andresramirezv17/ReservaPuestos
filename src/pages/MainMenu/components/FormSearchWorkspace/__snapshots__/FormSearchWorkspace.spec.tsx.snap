// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch test should match with snapshot 1`] = `
<div>
  <form
    autocomplete="off"
    class="makeStyles-root-2"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink="true"
        for="outlined-read-only-input"
        id="outlined-read-only-input-label"
      >
        Edificio
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="outlined-read-only-input"
          readonly=""
          type="text"
          value="Torre Central 2"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              Edificio
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root makeStyles-formControl-3"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink="true"
        id="demo-simple-select-outlined-label"
      >
        Piso
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          id="demo-simple-select-outlined"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              Piso
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root makeStyles-formControl-3"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink="true"
        id="demo-simple-select-outlined-label"
      >
        Sección
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          id="demo-simple-select-outlined"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
          >
            <span>
              Sección
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="makeStyles-boxDate-4"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        format="dd/MM/yyyy"
        style="margin-left: 13px;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Fecha
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            readonly=""
            type="text"
            value="14/09/2021"
          />
        </div>
        
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
        format="hh:mm a"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Hora inicio
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            readonly=""
            type="text"
            value="07:00 AM"
          />
        </div>
        
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
        format="hh:mm a"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Hora fin
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            readonly=""
            type="text"
            value="07:00 PM"
          />
        </div>
        
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btnSearch-5"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Buscar Puestos
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
